src/main/java/it/gov/pagopa/admissibility/service/onboarding/OnboardingContextHolderServiceImpl.java

```diff
@@ -150,10 +158,23 @@ public ApiKeysPDND getPDNDapiKeys(InitiativeConfig initiativeConfig) {
     }
 
     private ApiKeysPDND getApiKeysPDND(InitiativeConfig initiativeConfig) {
-        return ApiKeysPDND.builder()
-                .apiKeyClientId(aesTokenService.decrypt(initiativeConfig.getApiKeyClientId()))
-                .apiKeyClientAssertion(aesTokenService.decrypt(initiativeConfig.getApiKeyClientAssertion()))
-                .build();
+        try {
+            ApiKeysPDND apiKeysPDND = ApiKeysPDND.builder()
+                    .apiKeyClientId(aesTokenService.decrypt(initiativeConfig.getApiKeyClientId()))
+                    .apiKeyClientAssertion(aesTokenService.decrypt(initiativeConfig.getApiKeyClientAssertion()))
+                    .build();
+            apiKeysPDND.setAgidJwtTokenPayload(retrieveAgidTokenPayload(apiKeysPDND.getApiKeyClientAssertion()));
```

Comment:
due to the dependency between the clientAssertion field and the AgidJwtTokenPayload, we could move this logic inside the constructor (having all fields) of ApiKeysPDND class
In order to enforce the logic, we could set as final all the fields